test/Microsoft.AspNetCore.Mvc.Core.Test/PhysicalFileResultTest.cs

```diff
@@ -94,6 +232,53 @@ public void Constructor_SetsContentTypeAndParameters()
             sendFileMock.Verify();
         }
 
+        [Theory]
+        [InlineData(0, 3, "File", 4)]
+        [InlineData(8, 13, "Result", 6)]
+        [InlineData(null, 3, "File", 4)]
+        [InlineData(8, null, "ResultTestFile contents¡", 26)]
+        public async Task ExecuteResultAsync_CallsSendFileAsyncWithRequestedRange_IfIHttpSendFilePresent(long? start, long? end, string expectedString, long contentLength)
+        {
+            // Arrange
+            var path = Path.GetFullPath(Path.Combine("TestFiles", "FilePathResultTestFile.txt"));
+            var result = new TestPhysicalFileResult(path, "text/plain");
+
+            var sendFile = new TestSendFileFeature();
+            var httpContext = GetHttpContext();
+            httpContext.Features.Set<IHttpSendFileFeature>(sendFile);
+            var context = new ActionContext(httpContext, new RouteData(), new ActionDescriptor());
+            var requestHeaders = httpContext.Request.GetTypedHeaders();
+            requestHeaders.Range = new RangeHeaderValue(start, end);
+            requestHeaders.IfUnmodifiedSince = DateTimeOffset.Now;
```

Comment:
Strongly urge never to use `.Now` in a test because it literally changes all the time (well, it's *time* that changes, but let's not get into that!). Instead, always use some kind of computed date to guarantee the desired result. E.g. maybe look at the file's date, and add a year to it? Or whatever logic is appropriate?